datalake/backend/processing/check_files_checksums/task.py

```diff
@@ -1,8 +1,46 @@
 #!/usr/bin/env python3
-
 import sys
+from argparse import ArgumentParser
+from typing import Callable
+
+import boto3
+from botocore.response import StreamingBody
+from multihash import FUNCS, decode
+
+CHUNK_SIZE = 1024
+
+
+def validate_url_multihash(
+    url: str, hex_multihash: str, url_reader: Callable[[str], StreamingBody]
+) -> bool:
+    url_stream = url_reader(url)
+    checksum_function_code = int(hex_multihash[:2], 16)
+    checksum_function = FUNCS[checksum_function_code]
+
+    file_digest = checksum_function()
+    for chunk in url_stream.iter_chunks(chunk_size=CHUNK_SIZE):
+        file_digest.update(chunk)
+
+    return file_digest.digest() == decode(bytes.fromhex(hex_multihash))
+
+
+def parse_arguments():
+    argument_parser = ArgumentParser()
+    argument_parser.add_argument("--file-url", required=True)
+    argument_parser.add_argument("--hex-multihash", required=True)
+    arguments = argument_parser.parse_args()
+    return arguments
+
+
+def main() -> int:
+    arguments = parse_arguments()
+    s3_client = boto3.client("s3")
+
+    if validate_url_multihash(arguments.file_url, arguments.hex_multihash, s3_client):
+        return 0
+
+    return 1
 
-print("PAYLOAD:", sys.argv[1:])
 
-resp = {"item": "<FILE-NAME>", "success": True, "message": ""}
```

Comment:
Would you please add final result of check as `{"success": True, "message": ""}`